src/Core/Settings.h

```diff
@@ -641,7 +641,8 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Bool, compatibility_ignore_auto_increment_in_create_table, false, "Ignore AUTO_INCREMENT keyword in column declaration if true, otherwise return error. It simplifies migration from MySQL", 0) \
     M(Bool, multiple_joins_try_to_keep_original_names, false, "Do not add aliases to top level expression list on multiple joins rewrite", 0) \
     M(Bool, optimize_distinct_in_order, true, "Enable DISTINCT optimization if some columns in DISTINCT form a prefix of sorting. For example, prefix of sorting key in merge tree or ORDER BY statement", 0) \
-    M(Bool, optimize_sorting_by_input_stream_properties, true, "Optimize sorting by sorting properties of input stream", 0) \
+    M(Bool, optimize_sorting_by_input_stream_properties, true, "Optimize sorting by sorting properties of input stream", 0)                                                                                                           \
+    M(String, default_user_timezone, "", "Use specified timezone for interpreting Date and DateTime instead of server's timezone.", 0) \
```

Comment:
This is a query-level setting,
naming it "default_user_timezone" would be wrong.

Reply:
Maybe `force_timezone` ?

Reply:
Do we want to override ALL timezone-containig types? E.g. I have a column of `DateTime64(3, 'America/Denver')`, shall it be forcely made user tz? Or just substituting user-defined tz in case tz not explicitle provided in data

Reply:
No.

Reply:
No to what?

Reply:
This PR MUST NOT override any explicitly set timezones (either via column type, or timezone as function argument).
It also MUST NOT override default timezone for entire server.

Only operations/values with no explicit timezone should be affected, like:
`DateTime(xyz)`
`toDateTime('2016-06-15 23:00:00')`